packages/tx/src/baseTransaction.ts

```diff
@@ -46,7 +43,9 @@ interface TransactionCache {
  *
  * It is therefore not recommended to use directly.
  */
-export abstract class BaseTransaction<TransactionObject> {
+export abstract class BaseTransaction<TTransactionType extends TransactionType>
```

Comment:
```suggestion
export abstract class BaseTransaction<T extends TransactionType>
```
just using `T` should be sufficient and understandable for generics while keeping it short

Reply:
Good point, done.